database_test.go

```diff
@@ -263,45 +218,39 @@ func testSeekKeyValueIterator(db Database, t *testing.T) {
 	kv := testKVData()
 
 	it := db.NewIterator().(*BadgerIterator)
-	defer func() {
-		if it.Released() != true {
-			it.Release()
-		}
-	}()
+	defer it.Release()
 
 	for _, k := range kv {
-		k := k
-		t.Run("SeekKeyValueIterator", func(t *testing.T) {
-			it.Seek([]byte(k.input))
-			if !bytes.Equal(it.Key(), []byte(k.input)) {
-				t.Fatalf("failed to retrieve presented key, got %v, expected %v", it.Key(), k.input)
-			}
-			it.Seek([]byte(k.input))
-			if !bytes.Equal(it.Value(), []byte(k.expected)) {
-				t.Fatalf("failed to retrieve presented key, got %v, expected %v", it.Key(), k.expected)
-			}
-		})
+		it.Seek([]byte(k.input))
+		if !bytes.Equal(it.Key(), []byte(k.input)) {
+			t.Fatalf("failed to retrieve presented key, got %v, expected %v", it.Key(), k.input)
+		}
+		it.Seek([]byte(k.input))
+		if !bytes.Equal(it.Value(), []byte(k.expected)) {
+			t.Fatalf("failed to retrieve presented key, got %v, expected %v", it.Key(), k.expected)
+		}
 	}
 }
 
-func newTestBadgerDB(t *testing.T) (Database, func()) {
-	dir, err := ioutil.TempDir(os.TempDir(), "test_data")
+func newTestBadgerDB(t *testing.T) Database {
+	dir, err := ioutil.TempDir("", "test_data")
 	if err != nil {
 		t.Fatal("failed to create temp dir: " + err.Error())
 	}
+	t.Cleanup(func() {
+		if err := os.RemoveAll(dir); err != nil {
+			t.Error(err)
+		}
+	})
 
-	badgerDb, err := NewBadgerDB(dir)
+	db, err := NewBadgerDB(dir)
 	if err != nil {
 		t.Fatal(err)
 	}
-
-	db := Database(badgerDb)
-	return db, func() {
+	t.Cleanup(func() {
```

Comment:
this is very cool! is this from go 1.14?

Reply:
Yep! Will update the README, though this is a test requirement only.

Reply:
good point, maybe note that in the README